### ISSUE TITLE
Getting 500(Internal Error) for the graphs

### ISSUE THREAD

**Issue by ankush-grover-3pg** [Icinga/icingaweb2-module-graphite#20]
Hi,

I am running Icinga2 2.4.10 along with icingaweb2 2.4.1-1 on Centos 7. After enabling the graphite module I am getting below errors in Icinga related to graphite. The graphs are not shown on the Icinga Web Interface, please see the screenshot for the issue I am getting. If I access the graphite url on the browser I am able to see the graphs. This is a satellite host and Master Servers(2 in Nos) and another satellite host are showing the graphs.

![graphite](https://cloud.githubusercontent.com/assets/6574295/24033790/9c035a12-0b14-11e7-8109-9da4c1bf2d70.png)

Icinga Logs

Mar 15 05:59:48 icinga-ffx icingaweb2[26014]: ErrorException in /usr/share/icingaweb2/modules/graphite/library/Graphite/GraphiteChart.php:119 with message: file_get_contents(http://graphiteweb-ffx.example.com/render?height=150&amp;amp;width=300&amp;amp;_salt=1489571988.000&amp;amp;from=-1hours&amp;amp;hideLegend=1&amp;amp;hideGrid=true&amp;amp;vTitle=Percent&amp;amp;lineMode=connected&amp;amp;xFormat=%25a%20%25H%3A%25M&amp;amp;drawNullAsZero=false&amp;amp;graphType=line&amp;amp;tz=Europe%2FBerlin&amp;amp;title=AgilityRoomAppleTV-ffx&amp;amp;min=0&amp;amp;yUnitSystem=msec&amp;amp;areaMode=all&amp;amp;areaAlpha=0.4&amp;amp;target=alias%28color%28icinga2.AgilityRoomAppleTV-ffx.host.hostalive.perfdata.pl.value%2C%27red%27%29%2C%27Packet%20loss%27%29&amp;amp;target=alias%28color%28icinga2.AgilityRoomAppleTV-ffx.host.hostalive.perfdata.rta.value%2C%27%2344bb77%27%29%2C%27Round%20trip%20t        #0 [internal function]: Icinga\Application\ApplicationBootstrap->Icinga\Application\{closure}(2, 'file_get_conten...', '/usr/share/icin...', 119, Array)    #1 /usr/share/icingaweb2/modules/graphite/library/Graphite/GraphiteChart.php(119): file_get_contents('http://graphite...', false, Resource id #142)    #2 /usr/share/icingaweb2/modules/graphite/application/controllers/ShowController.php(144): Icinga\Module\Graphite\GraphiteChart->fetchImage()    #3 /usr/share/icingaweb2/library/vendor/Zend/Controller/Action.php(507): Icinga\Module\Graphite\Controllers\ShowController->graphAction()    #4 /usr/share/php/Icinga/Web/Controller/Dispatcher.php(76): Zend_Controller_Action->dispatch('graphAction')    #5 /usr/share/icingaweb2/library/vendor/Zend/Controller/Front.php(937): Icinga\Web\Controller\Dispatcher->dispatch(Object(Icinga\Web\Request), Object(Icinga\Web\Response))    #6 /usr/share/php/Icinga/Application/Web.php(389): Zend_Controller_Front->dispatch(Object(Icinga\Web\Request), Object(Icinga\Web\Response))    #7 /usr/share/php/Icinga/Application/webrouter.php(109): Icinga\Application\Web->dispatch()    #8 /usr/share/icingaweb2/public/index.php(4): require_once('/usr/share/php/...')    #9 {main}
Mar 15 06:00:35 icinga-ffx icingaweb2[26016]: ErrorException in /usr/share/icingaweb2/modules/graphite/library/Graphite/GraphiteChart.php:119 with message: file_get_contents(http://graphiteweb-ffx.example.com/render?height=150&amp;amp;width=300&amp;amp;_salt=1489572035.000&amp;amp;from=-1hours&amp;amp;hideLegend=1&amp;amp;hideGrid=true&amp;amp;vTitle=Percent&amp;amp;lineMode=connected&amp;amp;xFormat=%25a%20%25H%3A%25M&amp;amp;drawNullAsZero=false&amp;amp;graphType=line&amp;amp;tz=Europe%2FBerlin&amp;amp;title=AgilityRoomAppleTV-ffx&amp;amp;min=0&amp;amp;yUnitSystem=msec&amp;amp;areaMode=all&amp;amp;areaAlpha=0.4&amp;amp;target=alias%28color%28icinga2.AgilityRoomAppleTV-ffx.host.hostalive.perfdata.pl.value%2C%27red%27%29%2C%27Packet%20loss%27%29&amp;amp;target=alias%28color%28icinga2.AgilityRoomAppleTV-ffx.host.hostalive.perfdata.rta.value%2C%27%2344bb77%27%29%2C%27Round%20trip%20t        #0 [internal function]: Icinga\Application\ApplicationBootstrap->Icinga\Application\{closure}(2, 'file_get_conten...', '/usr/share/icin...', 119, Array)    #1 /usr/share/icingaweb2/modules/graphite/library/Graphite/GraphiteChart.php(119): file_get_contents('http://graphite...', false, Resource id #142)    #2 /usr/share/icingaweb2/modules/graphite/application/controllers/ShowController.php(144): Icinga\Module\Graphite\GraphiteChart->fetchImage()    #3 /usr/share/icingaweb2/library/vendor/Zend/Controller/Action.php(507): Icinga\Module\Graphite\Controllers\ShowController->graphAction()    #4 /usr/share/php/Icinga/Web/Controller/Dispatcher.php(76): Zend_Controller_Action->dispatch('graphAction')    #5 /usr/share/icingaweb2/library/vendor/Zend/Controller/Front.php(937): Icinga\Web\Controller\Dispatcher->dispatch(Object(Icinga\Web\Request), Object(Icinga\Web\Response))    #6 /usr/share/php/Icinga/Application/Web.php(389): Zend_Controller_Front->dispatch(Object(Icinga\Web\Request), Object(Icinga\Web\Response))    #7 /usr/share/php/Icinga/Application/webrouter.php(109): Icinga\Application\Web->dispatch()    #8 /usr/share/icingaweb2/public/index.php(4): require_once('/usr/share/php/...')    #9 {main}

Apache Logs

POST /render?height=150&amp;width=300&amp;_salt=1489572035.000&amp;from=-1hours&amp;hideLegend=1&amp;hideGrid=true&amp;vTitle=Percent&amp;lineMode=connected&amp;xFormat=%25a%20%25H%3A%25M&amp;drawNullAsZero=false&amp;graphType=line&amp;tz=Europe%2FBerlin&amp;title=AgilityRoomAppleTV-ffx&amp;min=0&amp;yUnitSystem=msec&amp;areaMode=all&amp;areaAlpha=0.4&amp;target=alias%28color%28icinga2.AgilityRoomAppleTV-ffx.host.hostalive.perfdata.pl.value%2C%27red%27%29%2C%27Packet%20loss%27%29&amp;target=alias%28color%28icinga2.AgilityRoomAppleTV-ffx.host.hostalive.perfdata.rta.value%2C%27%2344bb77%27%29%2C%27Round%20trip%20time%27%29&amp;_ext=whatever.svg HTTP/1.0" 500 2567
GET /icingaweb2/graphite/show/graph?template=icinga2%2Fping4&start=-1hours&width=300&height=150&hostname=expressway-us-e-192_168_21_105&service=ping4 HTTP/1.1" 500 3099
172.20.21.14 - - [15/Mar/2017:05:59:48 -0400] "GET /icingaweb2/graphite/show/graph?template=icinga2%2Fhostalive&start=-1hours&width=300&height=150&hostname=AgilityRoomAppleTV-ffx HTTP/1.1" 500 3048
172.20.21.14 - - [15/Mar/2017:06:00:35 -0400] "GET /icingaweb2/graphite/show/graph?template=icinga2%2Fhostalive&start=-1hours&width=300&height=150&hostname=AgilityRoomAppleTV-ffx HTTP/1.1" 500 3048

Thanks & Regards

Ankush Grover

**Comment by ankush-grover-3pg** [comment]
Any update on this issue? any suggestion where can be the issue for graphs not getting displayed

**Comment by datadavev** [comment]
I was getting this same error and was able to correct it by editing GraphiteChart.php. Here's the diff:

```diff
diff --git a/library/Graphite/GraphiteChart.php b/library/Graphite/GraphiteChart.php
index 4217d13..6654d7f 100644
--- a/library/Graphite/GraphiteChart.php
+++ b/library/Graphite/GraphiteChart.php
@@ -116,6 +116,7 @@ class GraphiteChart
         );

         $context = stream_context_create($options);
-        return file_get_contents($this->getUrl(), false, $context);
+        $url = html_entity_decode($this->getUrl());
+        return file_get_contents($url, false, $context);
     }
 }
```

**Comment by Al2Klimov** [comment]
@ankush-grover-3pg Has your problem been fixed w/ 777204d?

**Comment by bobapple** [comment]
I'm closing this due to lacking feedback
